src/features/game/types/milestones.ts

```diff
@@ -0,0 +1,146 @@
+    percentageComplete: (farmActivity: GameState["farmActivity"]) => {
+      const totalFishRequired = FISH_BY_TYPE.basic.length * 5;
+
+      const totalFishCaught = FISH_BY_TYPE.basic.reduce(
+        (total, name) =>
+          total + Math.min(farmActivity[`${name} Caught`] ?? 0, 5),
+        0
+      );
+
+      return Math.min((totalFishCaught / totalFishRequired) * 100, 100);
+    },
+    reward: {
+      "Sunflower Rod": 1,
+    },
+  },
+  "Advanced Angler": {
+    task: "Catch 5 of each advanced fish",
+    percentageComplete: (farmActivity: GameState["farmActivity"]) => {
+      const totalFishRequired = FISH_BY_TYPE.advanced.length * 5;
+
+      const totalFishCaught = FISH_BY_TYPE.advanced.reduce(
+        (total, name) =>
+          total + Math.min(farmActivity[`${name} Caught`] ?? 0, 5),
+        0
+      );
+
+      return Math.min((totalFishCaught / totalFishRequired) * 100, 100);
+    },
+    reward: {
+      "Fishing Hat": 1,
+    },
+  },
+  "Expert Angler": {
+    task: "Catch 5 of each expert fish",
+    percentageComplete: (farmActivity: GameState["farmActivity"]) => {
+      const totalFishRequired = FISH_BY_TYPE.expert.length;
+
+      const totalFishCaught = FISH_BY_TYPE.expert.reduce(
+        (total, name) =>
+          total + Math.min(farmActivity[`${name} Caught`] ?? 0, 5),
+        0
+      );
+
+      return Math.min((totalFishCaught / totalFishRequired) * 100, 100);
+    },
+    reward: {
+      "Angler Waders": 1,
+    },
+  },
+  "Fish Encyclopedia": {
+    task: "Discover all fish",
+    percentageComplete: (farmActivity: GameState["farmActivity"]) => {
+      const totalFishRequired = getKeys(FISH).length;
+
+      const totalFishCaught = getKeys(FISH).reduce(
+        (total, name) =>
+          total + Math.min(farmActivity[`${name} Caught`] ?? 0, 1),
+        0
+      );
+
+      return Math.min((totalFishCaught / totalFishRequired) * 100, 100);
+    },
+    reward: {
+      "Luminous Anglerfish Topper": 1,
+    },
+  },
+  "Master Angler": {
+    task: "Catch 10 of every fish",
+    percentageComplete: (farmActivity: GameState["farmActivity"]) => {
+      const totalFishRequired = getKeys(FISH).length * 10;
+
+      const totalFishCaught = getKeys(FISH).reduce(
+        (total, name) =>
+          total + Math.min(farmActivity[`${name} Caught`] ?? 0, 10),
+        0
+      );
+
+      return Math.min((totalFishCaught / totalFishRequired) * 100, 100);
+    },
+    reward: {
+      Trident: 1,
+    },
+  },
+};
+
+// All Milestones
+export const MILESTONES: Record<MilestoneName, Milestone> = {
+  ...FISH_MILESTONES,
+};
+
+export type ExperienceLevel = "Novice" | "Experienced" | "Expert";
+
+/**
+ * Helper function to get the experience level of the player for a specific codex category eg. Fishing
+ * @param claimed number of milestones claimed
+ * @param totalMilestones total number of milestones
+ * @returns ExperienceLevel
+ */
+export const getExperienceLevelForMilestones = (
```

Comment:
Cool concept.

Not an explicit "experience" but rather "experience implied by actions/accomplishments".

Maybe something similar could be applied to Cooking.  A simple example is that you might get a small cooking speed buff after you've passed some formula-based thresholds of "Cooked" analytics.  Somewhat related: This would make "cook your own food" a benefit for a bumpkin vs. "donated" food from tossing around bumpkins like hot potato.